src/service/ActivityEventService.ts

```diff
@@ -3,12 +3,28 @@ import { ActivityEvent } from "../model/ActivityEvent"
 import { ActivityEventRepository } from "../repository/ActivityEventRepository"
 import { SecurityContext, ActionContext, _verify } from "./Security"
 import jsonata from "jsonata"
-
+import Ajv from 'ajv';
+import api_schema from './api_schema.json'
+  
 export const ActivityEventService = Router()
 ActivityEventService.post("/participant/:participant_id/activity_event", async (req: Request, res: Response) => {
   try {
-    let participant_id = req.params.participant_id
+    let participant_id = req.params.participant_id    
+    var ajv = new Ajv({allErrors: true, unknownFormats: ['int64'], jsonPointers: false, useDefaults:false});
+    ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'));
+    let activity_schema = api_schema.components.schemas.ActivityEvent;
+    let schema_data:any = activity_schema;
+    schema_data.components = {
+      "schemas": {
+        "Timestamp": api_schema.components.schemas.Timestamp,
+        "Identifier": api_schema.components.schemas.Identifier
+      }
+    } 
+    let validate_schema = ajv.compile(schema_data);
     const activity_event = req.body
+    if(!(validate_schema(activity_event))){
+      res.status(500).json({ error: ajv.errorsText(validate_schema.errors)  })
+    }    
```

Comment:
Please follow proper code style and ensure you're using [Prettier](https://prettier.io) correctly.